### ISSUE TITLE
Add an option for requesting buffers of a specific size when building a stream

### ISSUE THREAD

**Issue by mitchmindtree** [RustAudio/cpal#205]
Currently the buffer size varies greatly between different backends. E.g. coreaudio seems to consistently provide buffers of 512 frames, whereas the alsa and wasapi backends can vary greatly between 250 and 1000+ frames.

Some backends provide an API for requesting buffers of a specific size. This can be useful for tweaking the trade-off between cpu usage and low-latency (often useful in pro-audio applications).

This could possibly be a 3rd argument to the `build_{input/output}_stream` methods, e.g.

```rust
pub enum BufferSize {
    Default,
    Fixed(usize),
}

let input = event_loop.build_input_stream(&device, &format, BufferSize::Default).unwrap();
let output = event_loop.build_output_stream(&device, &format, BufferSize::Fixed(128)).unwrap();
```

Alternatively perhaps a `buffer_size` field can be added to the `Format` type?

From what I can remember not all backends or devices can guarantee that they can provide buffers of arbitrary, fixed sizes, so we'd either have to emphasise that this is a "best-effort" API or add an intermediary buffer so that we could manually provide buffers of a specified fixed size in the case that a backend or device cannot.

**Comment by lpil** [comment]
I'm building an application that aims to have relatively low latency using ALSA. It'd be great for my use case if a small buffer size could be requested.

Thanks,
Louis

**Comment by RamiHg** [comment]
Would love for this feature to exist as well. It is currently the only thing blocking me from using CPAL.

**Comment by mrDIMAS** [comment]
Any news on when it will be added?

**Comment by mitchmindtree** [comment]
I plan to address this as a part of [this work](https://nannou.cc/posts/moss_grant_announce), I'm just waiting to hear back from Mozilla on the final part of the agreement. I'll report back here once I begin working on this issue!

**Comment by lpil** [comment]
Brilliant! Thank you

**Comment by dansgithubuser** [comment]
I'm eager to get this solved, happy to contribute if possible -- I think my main unknown is how to test across different platforms/hosts. Couldn't find a contribution policy -- what's your setup?

**Comment by Boscop** [comment]
Any update on this? :)

**Comment by mitchmindtree** [comment]
Yes, please see #401. @JoshuaBatty and I are busy over the next couple of days but will continue work on this toward the end of this week.

**Comment by mrDIMAS** [comment]
Any news?

**Comment by mitchmindtree** [comment]
Closed via #401. Thanks @JoshuaBatty!
